### PR TITLE
Sample springboot oauth2

### PR DESCRIPTION
**Pull request by hm-distro**

### CODE REVIEW

File: application-development/helloworld-springboot-auth/.classpath [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (bmenasha) [inline application-development/helloworld-springboot-auth/.classpath]: Can we remove this file? 

I don't think we should be committing any IDE configuration files.

File: application-development/helloworld-springboot-auth/.idea/helloworld-springboot-auth.iml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (bmenasha) [inline application-development/helloworld-springboot-auth/.idea/helloworld-springboot-auth.iml]: an we remove this file?

I don't think we should be committing any IDE configuration files.

File: application-development/helloworld-springboot-auth/.idea/misc.xml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (bmenasha) [inline application-development/helloworld-springboot-auth/.idea/misc.xml]: an we remove this file?

I don't think we should be committing any IDE configuration files.
thanks

File: application-development/helloworld-springboot-auth/.idea/modules.xml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (bmenasha) [inline application-development/helloworld-springboot-auth/.idea/modules.xml]: an we remove this file?

I don't think we should be committing any IDE configuration files.
thanks

File: application-development/helloworld-springboot-auth/.idea/preferred-vcs.xml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (bmenasha) [inline application-development/helloworld-springboot-auth/.idea/preferred-vcs.xml]: an we remove this file?

I don't think we should be committing any IDE configuration files.
thanks

File: application-development/helloworld-springboot-auth/.project [hunk 6]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (bmenasha) [inline application-development/helloworld-springboot-auth/.project]: an we remove this file?

I don't think we should be committing any IDE configuration files.
thanks

File: application-development/helloworld-springboot-auth/.settings/org.eclipse.core.resources.prefs [hunk 7]
```diff
@@ -0,0 +1,5 @@
+eclipse.preferences.version=1
```
Comment (bmenasha) [inline application-development/helloworld-springboot-auth/.settings/org.eclipse.core.resources.prefs]: remove this file?

I don't think we should be committing any IDE configuration files.
thanks

File: application-development/helloworld-springboot-auth/.settings/org.eclipse.jdt.core.prefs [hunk 8]
```diff
@@ -0,0 +1,5 @@
+eclipse.preferences.version=1
```
Comment (bmenasha) [inline application-development/helloworld-springboot-auth/.settings/org.eclipse.jdt.core.prefs]: remove this file?

I don't think we should be committing any IDE configuration files.
thanks

File: application-development/helloworld-springboot-auth/.settings/org.eclipse.m2e.core.prefs [hunk 9]
```diff
@@ -0,0 +1,4 @@
+activeProfiles=
```
Comment (bmenasha) [inline application-development/helloworld-springboot-auth/.settings/org.eclipse.m2e.core.prefs]: remove this file?

I don't think we should be committing any IDE configuration files.
thanks

File: application-development/helloworld-springboot-auth/pom.xml [hunk 10]
```diff
@@ -0,0 +1,85 @@
+<!--
+~ Copyright (c) 2017 Google Inc. All Rights Reserved.
```
Comment (bmenasha) [inline application-development/helloworld-springboot-auth/pom.xml]: You can remove the 

"All Rights Reserved."

and just use "Copyright (c) 2017 Google Inc."

File: application-development/helloworld-springboot-auth/pom.xml [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+    <maven.compiler.target>${java.version}</maven.compiler.target> <!-- REQUIRED -->
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <appengine.maven.plugin>1.3.1</appengine.maven.plugin>
+  </properties>
+
+  <dependencies>
+    <dependency>
+      <groupId>org.springframework.boot</groupId>
+      <artifactId>spring-boot-starter-web</artifactId>
+      <version>1.4.4.RELEASE</version>
```
Comment (bmenasha) [inline application-development/helloworld-springboot-auth/pom.xml]: the flex getting started example uses 1.5.8 (https://github.com/GoogleCloudPlatform/getting-started-java/blob/master/helloworld-springboot/pom.xml#L46)

Was I so late in reviewing this that they upgraded :)

File: application-development/helloworld-springboot-auth/pom.xml [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+      <version>2.0.11.RELEASE</version>
+    </dependency>
+  </dependencies>
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.springframework.boot</groupId>
+        <artifactId>spring-boot-maven-plugin</artifactId>
+        <version>1.3.3.RELEASE</version>
```
Comment (bmenasha) [inline application-development/helloworld-springboot-auth/pom.xml]: they also used maven plugin 1.5.7

File: application-development/helloworld-springboot-auth/src/main/appengine/app.yaml [hunk 13]
```diff
@@ -0,0 +1,30 @@
+# [START_EXCLUDE]
+# Copyright 2017 Google Inc. All Rights Reserved.
```
Comment (bmenasha) [inline application-development/helloworld-springboot-auth/src/main/appengine/app.yaml]: can remove All Rights Reserved.

Should also remove the  

# [START_EXCLUDE]
+# [END_EXCLUDE]

comments which I understand are used by the code that provides the code snippets on our documentation site

File: application-development/helloworld-springboot-auth/src/main/resources/application.yml [hunk 14]
```diff
@@ -0,0 +1,16 @@
+security:
+  oauth2:
+    client:
+      clientId: 667713450471-6u065o1ph801fr80kqa0popiudsvathb.apps.googleusercontent.com
+      clientSecret: cqR_pUlx1BjEBumoE0o69WkF
```
Comment (bmenasha) [inline application-development/helloworld-springboot-auth/src/main/resources/application.yml]: Should we remove this and just say <INSERT_YOUR_CLIENT_ID HERE>? or something? 

I assume that including this lets them run locally without creating a client id, but they will be unable to deploy and get the "invalid redirect url", right? 
I don't have a strong opinion.

File: application-development/helloworld-springboot-auth/src/main/resources/application.yml [hunk 15]
```diff
@@ -0,0 +1,16 @@
...
+      accessTokenUri: https://www.googleapis.com/oauth2/v4/token
+      userAuthorizationUri: https://accounts.google.com/o/oauth2/v2/auth
+      authenticationScheme: form
+      scope:
+        - openid
+        - email
+        - profile
+        - https://www.googleapis.com/auth/cloud-platform
```
Comment (bmenasha) [inline application-development/helloworld-springboot-auth/src/main/resources/application.yml]: should remove this "cloud-platform" scope as it's very wide and shouldn't e needed, and why openid?  I would think just email and profile is necessary.
thanks

File: application-development/helloworld-springboot-auth/src/main/resources/application.yml (hunk 14 above)
Comment (hm-distro) [reply application-development/helloworld-springboot-auth/src/main/resources/application.yml]: Thanks for catching this... I thought I had cleaned this up

**Comment by ryanmcdowell** [comment]
Closing as this is > 6 months old. Please re-open and complete review if still valid.

**Pull request closed without merging**